planar/triangulate/quadedge/quadedgesubdivision.go

```diff
@@ -478,7 +479,35 @@ Returns true if the vertex lies on the edge
 If qes is nil a panic will occur.
 */
 func (qes *QuadEdgeSubdivision) IsOnEdge(e *QuadEdge, p geom.Pointer) bool {
-	dist := planar.DistanceToLineSegment(p, e.Orig(), e.Dest())
+	// vertext version of DistanceToLineSegment from planar/distance.go
```

Comment:
The `planer` algorithms all call `XY()` and create new [2]float64 arrays. I've converted IsOnEdge, IsInCircle and EqualsToTolerance to directly use the Vertex.